docs/advanced-features/instrumentation.md

```diff
@@ -0,0 +1,37 @@
+---
+description: Learn how instrument your Next.js app.
+---
+
+> **Note**: This feature is experimental, you need to explicitly opt-in by providing `experimental.instrumentationHook = true;` in your `next.config.js`.
+
+# `instrumentation.ts`
+
+You can export `register` function from this file and we will be called whenever we bootstrap a new server instance.
+When deployed it will be called continuously on each coldboot (but exactly once in each environment).
+
+Many people used imports with side-effects before we released this feature.
```

Comment:
```suggestion
Sometimes, it may be useful to import a file in your code because of the side effects it will cause. For example, you might import a file that defines a set of global variables, but never explicitly use the imported file in your code. You would still have access to the global variables the package has declared.

```